src/storage/wal_replay.cpp

```diff
@@ -235,17 +231,19 @@ bool WriteAheadLog::Replay(AttachedDatabase &database, unique_ptr<FileHandle> ha
 			}
 		}
 	} catch (std::exception &ex) { // LCOV_EXCL_START
+		// exception thrown in WAL replay: rollback
+		con.Query("ROLLBACK");
 		ErrorData error(ex);
-		if (error.Type() != ExceptionType::SERIALIZATION) {
-			// FIXME: this should report a proper warning in the connection
-			Printer::PrintF("Exception in WAL playback: %s\n", error.RawMessage());
-			// exception thrown in WAL replay: rollback
+		// serialization failure means a truncated WAL
```

Comment:
Unsure what guarantees this, now or in the future, I think you can probably get a SerializationException from other cases (unsure how, but we throw SerializationException in many places), even though I don't see exactly a path to this happening.